src/librustc/mir/repr.rs

```diff
@@ -850,6 +851,16 @@ pub enum BinOp {
     Gt,
 }
 
+impl BinOp {
+    pub fn is_checkable(self) -> bool {
+        use self::BinOp::*;
+        match self {
+            Add | Sub | Mul | Shl | Shr => true,
```

Comment:
Well they're really "can be used with `CheckedBinaryOp`".

I don't think divergence is a valid concern here though. There aren't any new cases where we're going to check for overflow. I only added this method to make the logic in `as_rvalue` a bit clearer.